src/com/spikes2212/genericsubsystems/drivetrains/commands/TurnToGlobalAngleWithPID.java

```diff
@@ -0,0 +1,121 @@
+package com.spikes2212.genericsubsystems.drivetrains.commands;
+
+import com.spikes2212.genericsubsystems.drivetrains.TankDrivetrain;
+import com.spikes2212.utils.PIDSettings;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+
+import java.util.function.Supplier;
+
+/**
+ * This command is used to turn {@link TankDrivetrain} to a specific degree
+ * using wpilib's {@link PIDController}.
+ *
+ * @see TankDrivetrain
+ * @see PIDController
+ *
+ * @author Simon "C" Kharmatsky
+ */
+public class TurnToGlobalAngleWithPID extends Command {
+
+	private TankDrivetrain drivetrain;
+	private PIDSource source;
+	private Supplier<Double> setpointSupplier;
+	private PIDSettings settings;
+
+	private PIDController controller;
+
+	private double lastTimeOnTarget = 0;
+
+	/**
+	 * This constructs new {@link TurnToGlobalAngleWithPID} using {@link PIDSource},
+	 * {@link Supplier<Double>} for the setpoint and the {@link PIDSettings} for the
+	 * command
+	 * 
+	 * @param drivetrain
+	 *            the {@link TankDrivetrain} this command requires and moves
+	 * @param source
+	 *            the {@link PIDSource} that is used by the {@link PIDController} to
+	 *            get feedback about the robot's position
+	 * @param setpointSupplier
+	 *            {@link Supplier<Double>} for the setpoint of the
+	 *            {@link PIDController}
+	 * @param settings
+	 *            {@link PIDSettings} for this command
+	 */
+	public TurnToGlobalAngleWithPID(TankDrivetrain drivetrain, PIDSource source, Supplier<Double> setpointSupplier,
+			PIDSettings settings) {
+		requires(drivetrain);
+		this.drivetrain = drivetrain;
+		this.source = source;
+		this.setpointSupplier = () -> {
+			double setpoint = setpointSupplier.get();
+			setpoint = setpoint % 360;
+			if (Math.abs(setpoint - source.pidGet()) > 180)
+				setpoint -= 360;

```

Comment:
make this a function and use it again in execute (otherwise if the angle setPoint have changed it will not work)